### PR TITLE
Devops 33: Add unit tests for front-end component and service files

### PR DESCRIPTION
**Pull request by frenchal17**
Added unit tests for front-end component and service files

### CODE REVIEW

File: devops-ui/src/app/demo-page/demo-page.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,81 @@
...
+    let mockObject: demoObject = new demoObject();
+    mockObject.data = "Frenchie is crying, because Nate is spending too much money!";
+    return Promise.resolve(mockObject);
+  }
+}
 
 describe('DemoPageComponent', () => {
+  let injector: TestBed;
+  let service: MockDemoPageService;
   let component: DemoPageComponent;
```
Comment (RussianPrussian) [inline devops-ui/src/app/demo-page/demo-page.component.spec.ts]: we should remain the service and component to something more sensical. i.e. demoPageService, demoPageComponent

File: devops-ui/src/app/demo-page/demo-page.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,81 @@
...
-    fixture = TestBed.createComponent(DemoPageComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [DemoPageService, DemoPageComponent]
+    });
+    injector = getTestBed();
+    
+    service = injector.get(DemoPageService);
```
Comment (RussianPrussian) [inline devops-ui/src/app/demo-page/demo-page.component.spec.ts]: you've got to rename these guys, too.

File: devops-ui/src/app/demo-page/demo-page.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { TestBed, getTestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { DemoPageService } from './demo-page.service';
+import { demoObject } from '../core/demo-object.model';
+import { environment } from '../../environments/environment';
+
+describe('DemoPageService', () => {
+  let injector: TestBed;
+  let service: DemoPageService;
```
Comment (RussianPrussian) [inline devops-ui/src/app/demo-page/demo-page.service.spec.ts]: and here

File: devops-ui/src/app/demo-page/demo-page.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    service = injector.get(DemoPageService);
+    httpMock = injector.get(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should return response of getHello endpoint as String', () => {
+    service.callTestEndpoint()
```
Comment (RussianPrussian) [inline devops-ui/src/app/demo-page/demo-page.service.spec.ts]: and here

File: devops-ui/src/app/demo-page/demo-page.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  let promiseObject: demoObject = new demoObject();
+  promiseObject.data = "Frenchie is crying, because Nate is spending too much money!";
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [DemoPageService]
+    });
+    injector = getTestBed();
+    service = injector.get(DemoPageService);
```
Comment (RussianPrussian) [inline devops-ui/src/app/demo-page/demo-page.service.spec.ts]: and here

File: devops-ui/src/app/demo-page/demo-page.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            expect(result).toEqual(promiseObject.data);
+        });
+
+    const req = httpMock.expectOne(`${environment.apiUrl}/accounts/getHello`);
+    expect(req.request.method).toBe("GET");
+    req.flush(promiseObject);
+  });
+
+  it('should return response of getHello endpoint as demoObject', () => {
+    service.callTestEndpointPopulateObject()
```
Comment (RussianPrussian) [inline devops-ui/src/app/demo-page/demo-page.service.spec.ts]: you get the idea.

**Comment by frenchal17** [comment]
:facepalm:

**Pull request merged**